base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -643,7 +650,24 @@ where B: BlockchainBackend
     pub fn prepare_block_merkle_roots(&self, template: NewBlockTemplate) -> Result<Block, ChainStorageError> {
         let NewBlockTemplate { header, mut body, .. } = template;
         body.sort();
-        let header = BlockHeader::from(header);
+        let mut header = BlockHeader::from(header);
```

Comment:
This method now does more than it says it does. It should be renamed or this logic should be moved elsewhere

Reply:
I renamed the method to `prepare_new_block`. The timestamp was already set by this method, it was done inside of `let header = BlockHeader::from(header);`. All it does not is add an extra check to what it should set it and not just blanketly set it to now. But I do agree with the fact that the function name and what it does is misaligned.